src/instruments/hcp/tc038d.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env python
+"""
+Provides support for the TC038 AC crystal oven by HC Photonics.
+"""
+
+
+# IMPORTS #####################################################################
+
+
+from instruments.units import ureg as u
+
+from instruments.abstract_instruments.instrument import Instrument
+from instruments.util_fns import assume_units
+
+
+# CLASSES #####################################################################
+
+
+class TC038D(Instrument):
+    """
+    Communication with the HCP TC038D oven.
+
+    This is the newer version with DC heating.
+
+    The temperature controller is on default set to modbus communication.
+    The oven expects raw bytes written, no ascii code, and sends raw bytes.
+    For the variables are two or four-byte modes available. We use the
+    four-byte mode addresses, so do we. In that case element count has to be
+    double the variables read.
+    """
+
+    functions = {"read": 0x03, "writeMultiple": 0x10, "writeSingle": 0x06, "echo": 0x08}
+
+    byteMode = 4
+
+    def __init__(self, *args, **kwargs):
+        """
+        The TC038 is a crystal oven.
+
+        Example usage:
+
+        >>> import instruments as ik
+        >>> import instruments.units as u
+        >>> inst = ik.hcp.TC038.open_visa('TCPIP0:192.168.0.35')
+        >>> inst.setpoint = 45.3
+        >>> print(inst.temperature)
+        """
+        super().__init__(*args, **kwargs)
+        self.addr = 1
+
+    @staticmethod
+    def CRC16(data):
+        """Calculate the CRC16 checksum for the data byte array."""
+        CRC = 0xFFFF
+        for octet in data:
+            CRC ^= octet
+            for j in range(8):
+                lsb = CRC & 0x1  # least significant bit
+                CRC = CRC >> 1
+                if lsb:
+                    CRC ^= 0xA001
+        return [CRC & 0xFF, CRC >> 8]
+
+    def readRegister(self, address, count=1):
```

Comment:
Is `readRegister` and `writeMultiple` part of the user facing API for this instrument, or are they internal?

Reply:
Mostly they're used internally, but if someone wants to read or write registers, which I did not cover in the code, it's good to have these methods accessible. (You can change some more values, although I guess it might never happen, therefore I did not write the properties for them)